packages/legend-query/src/components/QueryBuilderFetchStructurePanel.tsx

```diff
@@ -79,7 +80,8 @@ export const QueryBuilderFetchStructurePanel = observer(
         if (fetchStructureState.fetchStructureMode !== fetchMode) {
           fetchStructureState.setFetchStructureMode(fetchMode);
           // TODO: might want to add alert modal to alert user changing fetch structure resets state
-          queryBuilderState.resetData();
+          queryBuilderState.resultSetModifierState =
```

Comment:
@MauricioUyaguari we should also reset the result modifier state here right? Potentially, `resetApp` because we would want to reset the result and all.

Reply:
As discussed with @akphi this is actually fine. We should only reset `resultSetModifierState`